packages/xo-server/src/xo-mixins/backups-ng/index.js

```diff
@@ -316,34 +327,24 @@ export default class BackupNg {
               return result
             } catch (error) {
               if (invalidParameters.is(error)) {
-                delete params.streamLogs
-                return app.callProxyMethod(proxyId, 'backup.run', params)
+                // wait for the result to properly reset backup listing cache in `finally`
+                return await app.callProxyMethod(proxyId, 'backup.run', params)
               }
               throw error
             }
           } else {
             const localTaskIds = { __proto__: null }
-            return await Task.run(
+            return await runBackupWorker(
               {
-                name: 'backup run',
-                onLog: log =>
-                  handleBackupLog(log, {
-                    localTaskIds,
-                    logger,
-                    runJobId,
-                  }),
+                config,
```

Comment:
Don't pass the whole `xo-server`'s config.

Define a clear API and pass only what's necessary.